internal/opencode/catalog_test.go

```
@@ -154,19 +213,377 @@ func TestDiscoverCatalogClassifiesCommandFailuresAndEmptyCatalog(t *testing.T) {
+func (e *exitOnCloseReader) WaitError() error           { return e.waitErr }
+
+// TestDiscoverCatalogPrefersCommandFailureOverParseClassification pins the
+// classification precedence of issue #4043 review feedback: a non-zero child
+// exit must surface as command_failed (or timeout) and must never be masked
+// by a downstream parse classification. Only a genuine output overflow keeps
+// its own category, matching the pre-streaming cmd.Run() behavior.
+func TestDiscoverCatalogPrefersCommandFailureOverParseClassification(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("requires a POSIX exit-status fixture")
+	}
+	fixture := exec.Command("sh", "-c", "exit 7")
```

golangci-lint (2.13.2)

[error] os/exec.Command must not be called. use os/exec.CommandContext

(noctx)